### DISCUSSION TITLE
Can You Use CloudScape with a Custom API on the Backend?

### DISCUSSION THREAD

**Discussion by stephmx** [cloudscape-design/components#1968] [Q&A] [answered]
Hello team. I have a custom API that returns information about our container images, and I would like to use some of CloudScape's resources to display said information to a user. The user needs to both GET information and PUT information to the API. Please let me know if this is possible. Thanks!

**Comment by connorlanigan** [comment, marked as answer]
Yes, you can use Cloudscape with any API you like. Cloudscape itself does not interact with or rely on any backend, it only provides components for the frontend.

For example, you can fetch a list of container images from your API in your application, and then display that data by placing it in Cloudscape's [Table component](https://cloudscape.design/components/table/). Or, for a details page, you could show information about the container image in a [Container element](https://cloudscape.design/components/container/) with a [Key-value pair layout](https://cloudscape.design/components/key-value-pairs/).
You can use Cloudscape's form-related components (e.g. [Form](https://cloudscape.design/components/form/), [FormField](https://cloudscape.design/components/form-field/), [Input](https://cloudscape.design/components/input/), [Select](https://cloudscape.design/components/select/), [Date picker](https://cloudscape.design/components/date-picker/), etc) to collect your user's data in a form, and when they click a [Button](https://cloudscape.design/components/button/), you can make a PUT request to your API.

We provide [some example pages on our website](https://cloudscape.design/demos/), and you can find the source code for each example by clicking the "Source code" link in the top right of each example page.
